pages/settings.php

```diff
@@ -69,6 +69,69 @@ function userSettings() {
             $system->message($e->getMessage());
         }
         $system->printMessage();
+    }
+	else if (!empty($_POST['upload_avatar'])) {
+		$valid_file_types = ['gif', 'png', 'jpg'];
+        try {
+			$target_dir = __DIR__ . '/../_user_imgs/';
+			$target_file = $target_dir . $system->db->clean(basename($_FILES['fileToUpload']['name']));
+			$file_type = strtolower(pathinfo($target_file, PATHINFO_EXTENSION));
+			$target_file_path = $target_dir . strtolower($player->user_name) . '.' . $file_type;
+			$user_link = $system->router->base_url . '_user_imgs/' . strtolower($player->user_name) . '.' . $file_type;
+			$current_file_path = null;
+
+			foreach($valid_file_types as $img_type) {
+				if(file_exists($target_dir . strtolower($player->user_name) . '.' . $img_type)) {
+					$current_file_path = $target_dir . strtolower($player->user_name) . '.' . $img_type;
+					break;
+				}
+			}
+			
+			// Limit uploads to 1/day
+			if($current_file_path && (time() - filemtime($current_file_path)) < 86400) {
+				throw new RuntimeException("You can only upload a file once per day!");
+			}
+
+			// Lack permission to upload file
+			if(!$player->forbidden_seal->direct_avatar_upload) {
+				throw new RuntimeException("You do not have permission to upload avatars!");
+			}
+	
+			// Upload directory is missing, this should be manually created
+            if(!is_dir($target_dir)) {
+                throw new RuntimeException("User upload directory is missing! Notify an administrator.");
+            }
+			// Only ally gif, png and jpg file types
+			if(!in_array($file_type, $valid_file_types)) {
```

Comment:
we probably need to look at how to inspect files to ensure no malware even if they're named correctly